core/src/androidMain/kotlin/com/revenuecat/purchases/kmp/Purchases.android.kt

```diff
@@ -200,7 +237,7 @@ public actual class Purchases private constructor(private val androidPurchases:
         }
             .build(),
         onError = { error, userCancelled -> onError(error.toPurchasesError(), userCancelled) },
-        onSuccess = { purchase, customerInfo -> onSuccess(purchase!!, customerInfo.toCustomerInfo()) },
+        onSuccess = { purchase, customerInfo -> onSuccess(purchase!!.toStoreTransaction(), customerInfo.toCustomerInfo()) },
```

Comment:
That's a good point... I'm not sure either but seems like this should never be null? cc @vegaro

Reply:
I got you. I think we used to have a null storeTransaction because of deferred upgrades, but we changed it and maybe forgot to update the lambda parameter in `purchaseWith`

```
fun Purchases.purchaseWith(
    purchaseParams: PurchaseParams,
    onError: (error: PurchasesError, userCancelled: Boolean) -> Unit = ON_PURCHASE_ERROR_STUB,
    onSuccess: (purchase: StoreTransaction?, customerInfo: CustomerInfo) -> Unit,
) {
    purchase(purchaseParams, purchaseCompletedCallback(onSuccess, onError))
}
```

```
internal fun purchaseCompletedCallback(
    onSuccess: (purchase: StoreTransaction, customerInfo: CustomerInfo) -> Unit,
    onError: (error: PurchasesError, userCancelled: Boolean) -> Unit,
)
```

I can't find the PR where that changed tho, but yes it feels like it should be non-null